pkgs/development/python-modules/bundlewrap/default.nix

```diff
@@ -5,58 +5,75 @@
   pythonOlder,
   cryptography,
   jinja2,
+  librouteros,
   mako,
+  packaging,
   passlib,
   pyyaml,
   requests,
   rtoml,
   setuptools,
   tomlkit,
-  librouteros,
   pytestCheckHook,
+  testers,
+  bundlewrap,
 }:
 
-buildPythonPackage rec {
-  pname = "bundlewrap";
-  version = "4.22.0";
-  pyproject = true;
+let
+  finalAttrs = {
+    pname = "bundlewrap";
+    version = "4.23.1";
+    pyproject = true;
 
-  disabled = pythonOlder "3.8";
+    src = fetchFromGitHub {
+      owner = "bundlewrap";
+      repo = "bundlewrap";
+      tag = finalAttrs.version;
+      hash = "sha256-Nzfx2L/FlYXQcbKq/cuRZ+PWnjv4HDld9q01nwQ1sA8=";
+    };
 
-  src = fetchFromGitHub {
-    owner = "bundlewrap";
-    repo = "bundlewrap";
-    tag = version;
-    hash = "sha256-F3Ipoep9ZmAqkh8mFLXpaEcYb4dpV9Dt/VgMa9X24Hw=";
-  };
+    build-system = [ setuptools ];
+    dependencies =
+      [
+        cryptography
+        jinja2
+        mako
+        passlib
+        pyyaml
+        requests
+        tomlkit
+        librouteros
+      ]
+      ++ lib.optionals (pythonOlder "3.11") [
+        rtoml
+      ]
+      ++ (if (pythonOlder "3.10") then [ setuptools ] else [ packaging ]);
+
+    pythonImportsCheck = [ "bundlewrap" ];
+
+    nativeCheckInputs = [ pytestCheckHook ];
+
+    pytestFlags = [
+      # only unit tests as integration tests need a OpenSSH client/server setup
+      "tests/unit"
+    ];
+
+    passthru.tests = {
+      version = testers.testVersion {
+        package = bundlewrap;
+        command = "bw --version";
+        inherit (finalAttrs) version;
```

Comment:
No need for `finalAttrs`. You can use `versionCheckHook`.
Please don't do the `finalAttrs` trick, we'll wait until `buildPythonPackage` supports it properly.

Reply:
Thanks for pointing this out. PR updated.